pkg/cli/admin/inspect/namespace.go

```diff
@@ -92,3 +101,65 @@ func (o *InspectOptions) gatherNamespaceData(baseDir, namespace string) error {
 	}
 	return nil
 }
+
+// podEligibleForCollection aims to filter out the daemonset pods running on
+// data planes if node count is greater than a threshold,
+// in order to save significant storage size and reduces the collection duration.
+// If these pods need to be collected, recommended path is
+// to use node-selector flag (i.e. --node-selector='node-role.kubernetes.io/worker').
+func (o *InspectOptions) podEligibleForCollection(pod *corev1.Pod) (bool, error) {
+	isDaemonSet := false
+	if len(pod.OwnerReferences) > 0 {
+		for _, OwnerRef := range pod.OwnerReferences {
+			if OwnerRef.Kind == "DaemonSet" {
+				isDaemonSet = true
+				break
+			}
+		}
+	}
+	if !isDaemonSet {
+		return true, nil
+	}
+
+	nodes, err := o.kubeClient.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return true, err
+	}
+	const daemonsetCollectionMaxNodeSize = 20
```

Comment:
Would it be 10 better?

Reply:
Note: it's ok to keep this as a non-configurable limit until we get complains/requests to change the value. At which point we can turn it into a new flag.